### PR TITLE
support for alosStack in ISCE

### PR DESCRIPTION
**Pull request by CunrenLiang**
**Description of proposed changes**

This PR adds support for ALOS-2 (and ALOS-4 in the future) stack processed with `contrib/stack/alosStack` in ISCE-2 (https://github.com/isce-framework/isce2/pull/175). The data loading part is similar to the product from topsStack and stripmapStack, whose sub-type is identified automatically based on the input metadata file via `mintpy.load.metaFile`.

Detailed changes are as follows:

+ utils/isce_utils.py:
   - modify get_processor() to detect `alosStack` based on `f1_*/*.frame.xml` file.
   - add the following new functions to extract metadata for alosStack:
        - extract_alosStack_metadata()
        - alos2_acquisition_mode()
        - extract_image_size_alosStack()
        - load_track()
        - read_alosStack_baseline()

+ prep_isce.py:
   - adjust prepare_stack() and prepare_geometry() for alosStack.
   - cmd_line_parse(): 1) use argparse required argument to replace the checking in cmd_line_parse(); 2) translate wildcard in input metaFile

+ add `docs/examples/input_files/NCalAlos2Scan.txt` as the example template file for loading alosStack products

+ objects/stackDict.geometry.write2hdf5(): convert waterBody to waterMask while loading data

+ utils/readfile.read_binary_file(): use band2 by default for *.cor file with 2 bands to support reading complex coherence files generated by isce-2.

**Reminders**

- [x] Pass Codacy code review (green)
- [x] Pass Circle CI test (green)
- [x] Make sure that your code follows our style. Use the other functions/files as a basis.
- [x] If modifying functionality, describe changes to function behavior and arguments in a comment below the function declaration.
- [x] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

**Comment by yunjunz** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mintpy/utils/isce_utils.py  3
- mintpy/prep_isce.py  2
- mintpy/load_data.py  1
- mintpy/objects/stackDict.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/insarlab/MintPy/pullRequest?prid=6380573&bid=20696837)

**Comment by yunjunz** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mintpy/objects/stackDict.py  2
- mintpy/utils/isce_utils.py  3
- mintpy/load_data.py  1
- mintpy/prep_isce.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/insarlab/MintPy/pullRequest?prid=6380573&bid=20696837)

**Comment by yunjunz** [comment]
Related to https://github.com/isce-framework/isce2/pull/175

File: mintpy/utils/isce_utils.py [hunk 1]
```diff
@@ -333,6 +346,159 @@ def extract_stripmap_metadata(meta_file):
...
+                    sensingMid.minute * 60.0 +
+                    sensingMid.second)
+    meta['CENTER_LINE_UTC'] = time_seconds
+
+    peg = track.orbit.interpolateOrbit(sensingMid, method='hermite')
+    Vs = np.linalg.norm(peg.getVelocity())
+    meta['azimuthPixelSize'] = Vs * track.azimuthLineInterval
+    meta['rangePixelSize'] = track.rangePixelSize
+
+    azb = track.prf * 0.8
```
Comment (yunjunz) [inline mintpy/utils/isce_utils.py:383]: Could you write the full name for this variable `azb` in the comments for better understanding?

**Comment by yunjunz** [comment]
This is awesome @CunrenLiang!

**Comment by yunjunz** [comment]
I have some very minor suggestions, it will be much easier for me to edit on the pull request directly, could you give me the permission @CunrenLiang?

Here is the instruction from GitHub for that: https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

File: mintpy/utils/isce_utils.py (hunk 1 above)
Comment (CunrenLiang) [reply mintpy/utils/isce_utils.py:383]: that is azimuth bandwidth

**Comment by CunrenLiang** [comment]
> I have some very minor suggestions, it will be much easier for me to edit on the pull request directly, could you give me the permission @CunrenLiang?
> 
> Here is the instruction from GitHub for that: https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

OK, you should have the permission to edit now.

**Comment by CunrenLiang** [comment]
Thanks for the detailed comments about the changes @yunjunz

File: mintpy/utils/isce_utils.py (hunk 1 above)
Comment (yunjunz) [reply mintpy/utils/isce_utils.py:383]: I see, thanks.

**Comment by yunjunz** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mintpy/objects/stackDict.py  2
- mintpy/utils/isce_utils.py  3
- mintpy/prep_isce.py  2
- mintpy/load_data.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/insarlab/MintPy/pullRequest?prid=6380573&bid=20696837)

**Review by yunjunz — APPROVED** [review]
This looks great! Thank you @CunrenLiang.

**Pull request merged**